src/idl/src/annotation.c

```diff
@@ -187,47 +183,61 @@ annotate_extensibility(
   idl_annotation_appl_t *annotation_appl,
   idl_node_t *node)
 {
-  idl_extensibility_t extensibility = IDL_EXTENSIBILITY_FINAL;
+  const idl_annotation_appl_t **annotationp = NULL;
+  idl_extensibility_t *extensibilityp = NULL, extensibility = IDL_FINAL;
   const char *annotation;
 
   assert(annotation_appl);
   annotation = idl_identifier(annotation_appl->annotation);
   assert(annotation);
 
   if (strcmp(annotation, "final") == 0) {
```

Comment:
That can't happen. Or rather, the syntax and semantics of the annotation is checked before that function is called. Of course, there's always the possibility of bugs but at some point I must have some confidence that the mechanism works as expected. Should that happen though, both pointers are `NULL` and the function returns `IDL_RETCODE_SEMANTIC_ERROR` resulting in parser failure.

Reply:
Correction, the pointers won't be `NULL`, but `extensibility` will have the value `IDL_FINAL`. Which might result in the struct, union or enum having the wrong "extensibility". However, if that situation arises, the real bug is that the invalid input hasn't been caught before this function is called.

Reply:
Very good; I was "led astray", or so one might say, by all the string compares with literals. I suppose you failed to find a good way to avoid having copies of the same keywords all over the place?

It does seem like a simple `else assert(0);` at the end of the if/else if/else if/... would at least provide some hint that the validation of the annotation strings happens at an earlier stage.

Reply:
For annotations they're not defined anywhere else except for the built-in annotations. Which is simply a string literal with a single annotation declaration and a specified callback that's called when an annotation application is encountered. Annotation application syntax is checked against the tree itself. I suppose I could split that out further, but I felt that'd only complicate things. Especially because annotations don't introduce keywords. I guess, I could define `FINAL` to be `"FINAL"` and use that in both places, but to verify if things work, I'm inclined to say tests add more in this case.